integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -124,15 +125,23 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
   this.timeout(60_000); // TODO: Temporarily hardcoded until envs are set up.
   importAppBefore("with-db-flx");
   authenticateUserBefore();
-  openRealmBeforeEach({
-    schema: [FlexiblePersonSchema, DogSchema],
-    sync: {
-      flexible: true,
-    },
-  });
 
   describe("Configuration", function () {
     describe("flexible sync Realm config", function () {
+      it.skip("respects cancelWaitOnNonFatalError", async function () {
+        this.timeout(2_000);
+        //This needs to be revisited when:
+        //- https://github.com/realm/realm-core/issues/6301 is fixed. At the moment core rejects all the "non sensible" timeouts, and we need to set a very low one to simulate reliably a connection issue
+        //- We expose `SyncClientConfig.timeouts`(https://github.com/realm/realm-core/blob/e83515f014ff53b0d27c59b73773d33b4252e891/src/realm/object-store/sync/sync_manager.hpp#L97) so that we can set the timeouts properly.
+        const config = createSyncConfig({ sync: { cancelWaitsOnNonFatalError: true, flexible: true } }, this.user);
+
+        const openRealm = async () => {
+          await Realm.open(config);
+        };
+
+        await expect(openRealm()).to.be.rejected;
```

Comment:
I think that the message here is not the important part. I've decided to keep it out also because the message itself could change when all the preconditions I've talked about are satisfied and we decide to test this properly. At the moment this is more of a "reminder" to add a test at some point when we can actually do it properly.

Reply:
Alright :) Was thinking more so that we can be more certain that it's throwing because of the right reason. But because of several preconditions like you said, it may be harder.